### PR TITLE
[Linux][GDB-JIT] Added DWARF exprloc for static value class fields

### PR DESCRIPTION
**Pull request by Buyduck**
Added DWARF exprloc for static value class fields.
Address for value class static fields should be calculated by (as it wrote in corinfo.h):
`addr = (*addr+sizeof(OBJECTREF))`

With this PR LLDB shows values of static structures correctly.

For example:
```
(lldb) p *C
(__MyClass) $0 = {__MyClass} {
  j = {Struct1} {
    x = 20
    y = 4.1415926535000001
  }
  Static members = {__MyClass} {
    i (__MyClass) = {Struct1} {
      x = 10
      y = 3.1415926535000001
    }
  }
}
```

@janvorli PTAL

cc @Dmitri-Botcharnikov @lucenticus @ayuckhulk

### CODE REVIEW

File: src/vm/gdbjit.cpp [hunk 1]
```diff
@@ -1154,19 +1155,47 @@ void TypeMember::DumpStaticDebugInfo(char* ptr, int& offset)
...
         memcpy(ptr + offset, &memberEntry, sizeof(DebugInfoStaticMember));
 
-        char buf[ptrSize + 2] = {0};
-        buf[0] = ptrSize + 1;
-        buf[1] = DW_OP_addr;
-
-        for (int i = 0; i < ptrSize; i++)
+        // for value type static fields compute address as:
+        // addr = (*addr+sizeof(OBJECTREF))
+        if (m_member_type->GetTypeHandle().GetSignatureCorElementType() ==
```
Comment (janvorli) [inline src/vm/gdbjit.cpp]: I am not sure if this is the case for all static fields of value types, based on the comments in corinfo.h (https://github.com/dotnet/coreclr/blob/master/src/inc/corinfo.h#L117-L154).
@jkotas, can you please comment on this?

Comment (jkotas) [inline src/vm/gdbjit.cpp]: The different types of fields are orthogonal problem. There is `TODO: add support of generics with static fields` in this file to handle them.

This change looks fine to me.

**Comment by jkotas** [comment]
@dotnet-bot test this please

**Comment by Buyduck** [comment]
@dotnet-bot help

**Comment by Buyduck** [comment]
@dotnet-bot test this please

**Comment by janvorli** [comment]
@dotnet-bot test Tizen armel Cross Debug Build please

**Comment by Buyduck** [comment]
@janvorli Thank you!

**Pull request merged**